### PR TITLE
Optimize CDP Begin Blocker

### PR DESCRIPTION
**Pull request by nddeluca**
Shows a ~10x improvement on mainnet and tested on blocks 1 to ~4.8 million

### CODE REVIEW

File: x/cdp/keeper/interest.go [hunk 1]
```diff
@@ -161,11 +162,110 @@ func (k Keeper) CalculateNewInterest(ctx sdk.Context, cdp types.CDP) sdk.Coin {
...
+	for ; iterator.Valid(); iterator.Next() {
+		_, id, _ := types.SplitCollateralRatioKey(iterator.Key())
+		cdpIDs = append(cdpIDs, id)
+		if int64(len(cdpIDs)) >= cp.CheckCollateralizationIndexCount.Int64() {
+			break
+		}
+	}
+	iterator.Close()
+
+	globalInterestFactor, found := k.GetInterestFactor(ctx, cp.Type)
```
Comment (nddeluca) [inline x/cdp/keeper/interest.go:183]: The previous version created this if it didn't exist, but the accumulate interest functions called before this method already performed that logic

Comment (nddeluca) [reply x/cdp/keeper/interest.go:183]: Looking into why this affected the eip712 test

Comment (nddeluca) [reply x/cdp/keeper/interest.go:183]: Resolved -- forgot to address the case when both previous accural & global interest factor are not set, but this is added as an abci test now

**Review by galxy25 — APPROVED** [review]
nice optimizations, I knew there was some low hanging fruit here to pluck with a little bit of profiling and tweaking - https://kava-labs.atlassian.net/wiki/spaces/ENG/pages/1362001933/First+Round+Performance+Optimizations+Design+Doc#Option-1 #TODONE

File: x/cdp/keeper/interest.go [hunk 2]
```diff
@@ -161,11 +162,111 @@ func (k Keeper) CalculateNewInterest(ctx sdk.Context, cdp types.CDP) sdk.Coin {
...
+	if !found {
+		globalInterestFactor = sdk.OneDec()
+		k.SetInterestFactor(ctx, cp.Type, globalInterestFactor)
```
Comment (nddeluca) [inline x/cdp/keeper/interest.go:186]: @rhuairahrighairidh returning here might be better? Though, I don't believe there is a case where a cdp can exist without a global interest factor set (create cdp sets it), so regardless believe it to be compatible with old behavior

Comment (rhuairahrighairidh) [reply x/cdp/keeper/interest.go:186]: yeah slight lean to just returning. I don't think it matters but not having interest sync (from a early return) feels not as bad as potentially incorrectly setting a global factor

**Review by rhuairahrighairidh — APPROVED** [review]

**Comment by nddeluca** [comment]
Moving to draft for some touch ups and to align better with current functionality

**Review by pirtleshell — APPROVED** [review]

**Review by galxy25 — APPROVED** [review]

**Comment by nddeluca** [comment]
Merging to master, but not back-porting to v0.26.x until after release

**Pull request merged**